Add basis for content translation (#1482)
vivid-planet/comet: .changeset/rare-shiny-bulbasaur.md

---
"@comet/admin": minor
---

Add the TranslationConfigProvider which can be used to wrap components that use FinalFormInput (and later also RTE) to enable the activation of the translation feature.

> ```ts

Reviewer: ```suggestion
```tsx
```

---
"@comet/admin": minor
---

> Add the ContentTranslationServiceProvider which can be used to wrap components that use FinalFormInput (and later also RTE) to enable the activation of the translation feature.

Reviewer: Add
```suggestion
Add basis for content translation

Wrap a component with a `ContentTranslationServiceProvider` to add support for content translation to all underlying `FinalFormInput` inputs. 
```

```tsx
<ContentTranslationServiceProvider
    enabled={true}
    translate={async function (input: string): Promise<string> {
        return yourTranslationFnc(value);

>     }}

Reviewer: This could wouldn't work, because you're using input and value. Change both to text please

>
    ...
</ContentTranslationServiceProvider>
```

> You can hide the translation button on the FinalFormInput by adding the disableTranslation prop.

Reviewer: ```suggestion
You can disable translation for a specific  `FinalFormInput` by using the `disableTranslation` prop.
```

```tsx
<Field
    required
    fullWidth
    name="myField"
    component={FinalFormInput}

>     disableTranslation

Reviewer: Maybe use ```diff to highlight the added line?

Author: I moved it to the bottom of the props and added a empty line as divider.

```tsx
<Field
    required
    fullWidth
    name="myField"
    component={FinalFormInput}    
    label={<FormattedMessage id="myField" defaultMessage="My Field" />}

    disableContentTranslation
/>
```

---
"@comet/admin": minor
---

Add basis for content translation

Wrap a component with a `ContentTranslationServiceProvider` to add support for content translation to all underlying `FinalFormInput` inputs.

```tsx
<ContentTranslationServiceProvider
    enabled={true}
    translate={async function (text: string): Promise<string> {
        return yourTranslationFnc(text);
    }}
>
    ...
</ContentTranslationServiceProvider>
```

You can disable translation for a specific `FinalFormInput` by using the `disableContentTranslation` prop.

> ```tsx
> <Field
>     required
>     fullWidth
>     name="myField"
>     component={FinalFormInput}    
>     label={<FormattedMessage id="myField" defaultMessage="My Field" />}
> 
>     disableContentTranslation

Reviewer 2: I think what @johnnyomair meant is:

```suggestion
```diff
<Field
    required
    fullWidth
    name="myField"
    component={FinalFormInput}    
    label={<FormattedMessage id="myField" defaultMessage="My Field" />}
+. disableContentTranslation
```

resulting in 

```diff
<Field
    required
    fullWidth
    name="myField"
    component={FinalFormInput}    
    label={<FormattedMessage id="myField" defaultMessage="My Field" />}
+   disableContentTranslation
/>
```
